src/models/implementations/presentable-model.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Nectary\Models;
+
+use Nectary\Model;
+
+/**
+ * A Viewable model is one that can present itself so that all of its
+ * properties and calculated properties are useable within a template
+ */
+abstract class Presentable_Model extends Model {
+  /**
+   * Present the model using the class provided
+   * by presents(). Present knows how to build
+   * factories.
+   *
+   * @return array
+   */
+  public function present( $options = [] ) {
+    $class_reference = $this->get_presenter();
+
+    if ( ! is_null( $class_reference ) ) {
+      $instance = new $class_reference( $this, $options );
+
+      if ( $instance instanceof Factory ) {
+        return $instance->build();
+      } else {
+        return to_array( $instance );
+      }
+    } else {
+      return to_array( $this );
+    }
+  }
+
+  /**
+   * Called to determine which class knows how to present the model.
+   * Returning the string of a Factory class is prefered.
+   *
+   * @return String The name of the class that knows how to present the model
+   */
+  abstract protected function get_presenter();
```

Comment:
@rraub This can basically be anything we want to call it. I'm leaning towards `get_presenter_class_name`. I think that makes it very clear what it should return.